tests/test_gc.rs

```diff
@@ -71,7 +81,9 @@ fn data_is_dropped() {
         drop(inst);
     });
 
+    #[cfg(not(Py_GIL_DISABLED))]
     assert!(drop_called1.load(Ordering::Relaxed));
+    #[cfg(not(Py_GIL_DISABLED))]
     assert!(drop_called2.load(Ordering::Relaxed));
```

Comment:
This test doesn't rely on a `__clear__` function, so I think shouldn't be flaky with gc. Was this change out of an abundance of caution or because you did observe flakiness?